src/index.js

```diff
@@ -90,7 +90,7 @@ options.vnode = vnode => {
 			if (vnode.children) attrs.children = vnode.children;
 
 			if (attrs.defaultValue) {
-				if (!attrs.value && attrs.value!==0) {
+				if (!attrs.value && attrs.value!==0 && attrs.value!=='') {
```

Comment:
What if we used this check instead of all three?

```js
if (attrs.value==null) {
  attrs.value = attrs.defaultValue;
}
```

Reply:
IMO, only `undefined` should be ignored.
```js
if (attrs.defaultValue) {
	if (attrs.value === undefined) {
		attrs.value = attrs.defaultValue;
	}
	delete attrs.defaultValue;
}
```

Reply:
actually, `in` check will be most strictly, but, preact-compat will merge defaultValue to value.

Reply:
What is React's behavior here?  They usually differentiate between `null` and `undefined`, but I haven't tested this specific behavior.  We should just use whatever they use to avoid an inconsistency.

Reply:
I also ran into problems with the `preact-compat` implementation of `defaultValue`.

In React, the `defaultValue` prop will not set the `value` prop on an element in general. In my observations, the `value` prop remains `undefined`. Changing the `defaultValue` prop does not cause any further changes/renders on the component.

Reply:
The described behavior applies to the case of a `select`.